nix/runtime.nix

```diff
@@ -0,0 +1,64 @@
+{
+  # Plumbing tools:
+  closureInfo, runCommand, writeText, buildEnv,
+
+  # Actual dependencies to propagate:
+  bash, coreutils }:
+let
+  tools = buildEnv {
+    name = "lorri-runtime-tools";
+    paths = [ coreutils bash ];
+  };
+
+  runtimeClosureInfo = closureInfo {
+    rootPaths = [ tools ];
+  };
+
+  runtimeClosureInfoAsNix = runCommand "runtime-closure.nix" {}
+  ''
+    cat <<EOF > scratch.nix
+    # Nix with sandboxing requires every path used at build time be
+    # explicitly declared. If we simply passed in the paths, they
+    # would be copied in as sources. Using builtins.storePath we're
+    # able to tell Nix that, no, in fact, treat these not as sources
+    # to copy, but instead of a regular store path.
+    #
+    # Include the explicit closure, too, otherwise we'll get mysterious
+    # "file not found" errors due to the glibc interpreter being
+    # missing.
+    let
+      # Magic inspired by Nix's config.nix:
+      # https://github.com/NixOS/nix/blob/f9a2ea44867cd1dbb408bca4df0ced806137b7f7/corepkgs/config.nix.in#L23
+      #
+      # If the dependency is in the Nix store we're using, refer to
+      # it as a literal store path. If it isn't, refer to it "normally".
+      #
+      # This makes sandboxing happy when in a nix-build, and the
+      # evaluation happy when in a «cargo build».
+      tools_host = ${tools};
+      pp = x: builtins.trace x x;
+
+      # Compare the stringified version of the tools_host Nix store path
+      # to the evaluator's stringified Nix store path. Otherwise,
+      # Nix will read the sources in to the /nix/store, and, well,
+      # you can only copy the /nix/store in to the /nix/store so many
+      # times before you run out of disk space.
+      dep = if ("\''${toString (dirOf tools_host)}" == "\''${toString builtins.storeDir}")
+        then (builtins.trace "using storePath"  builtins.storePath)
+        else (builtins.trace "using toString" toString) # assume we have no sandboxing
+        ;
+
+      tools = dep tools_host;
+
+    in {
+      path = "\''${tools}/bin";
+      builder = "\''${tools}/bin/bash";
+      closure = [
+    $(sed -E 's/^(.*)$/    (dep \1)/' ${runtimeClosureInfo}/store-paths)
+      ];
+    }
+    EOF
+
+    mv ./scratch.nix $out
+  '';
```

Comment:
Example output of this:

```nix
# Nix with sandboxing requires every path used at build time be
# explicitly declared. If we simply passed in the paths, they
# would be copied in as sources. Using builtins.storePath we're
# able to tell Nix that, no, in fact, treat these not as sources
# to copy, but instead of a regular store path.
#
# Include the explicit closure, too, otherwise we'll get mysterious
# "file not found" errors due to the glibc interpreter being
# missing.
let
  # Magic inspired by Nix's config.nix:
  # https://github.com/NixOS/nix/blob/f9a2ea44867cd1dbb408bca4df0ced806137b7f7/corepkgs/config.nix.in#L23
  #
  # If the dependency is in the Nix store we're using, refer to
  # it as a literal store path. If it isn't, refer to it "normally".
  #
  # This makes sandboxing happy when in a nix-build, and the
  # evaluation happy when in a «cargo build».
  tools_host = /nix/store/7yp1kp9m2s5cjklxj4l5rr9gcizfbd06-lorri-runtime-tools;
  pp = x: builtins.trace x x;

  # Compare the stringified version of the tools_host Nix store path
  # to the evaluator's stringified Nix store path. Otherwise,
  # Nix will read the sources in to the /nix/store, and, well,
  # you can only copy the /nix/store in to the /nix/store so many
  # times before you run out of disk space.
  dep = if ("${toString (dirOf tools_host)}" == "${toString builtins.storeDir}")
    then (builtins.trace "using storePath"  builtins.storePath)
    else (builtins.trace "using toString" toString) # assume we have no sandboxing
    ;

  tools = dep tools_host;

in {
  path = "${tools}/bin";
  builder = "${tools}/bin/bash";
  closure = [
    (dep /nix/store/4apxd9wcaamlss78hkv1bpm8yn1crfhb-bash-4.4-p23-man)
    (dep /nix/store/681354n3k44r8z90m35hm8945vsp95h1-glibc-2.27)
    (dep /nix/store/7yp1kp9m2s5cjklxj4l5rr9gcizfbd06-lorri-runtime-tools)
    (dep /nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23)
    (dep /nix/store/d9s1kq1bnwqgxwcvv4zrc36ysnxg8gv7-coreutils-8.30)
    (dep /nix/store/hql9ki8x230src24ljb9fad7rxxpzal0-attr-2.4.48)
    (dep /nix/store/s83xl21h4qiz5yxlhr0n2bm1dl574mhw-acl-2.2.53)
  ];
}
```